e2e/hook/internal/config.go

```diff
@@ -0,0 +1,33 @@
+package internal
+
+import (
+	"os"
+
+	"k8s.io/client-go/rest"
+)
+
+const envVarUseInsecure string = "E2E_USE_INSECURE_TLS"
+
+func setTLSFromEnv(cfg *rest.Config) {
+	if os.Getenv(envVarUseInsecure) != "false" {
```

Comment:
I'd suggest to change this to checking for explicit request of the feature

```suggestion
	if os.Getenv(envVarUseInsecure) == "true" {
```